.github/workflows/ci.yml

```
@@ -0,0 +1,38 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  unit-test:
+    runs-on: ubuntu-22.04
+    concurrency:
+      group: ${{ github.workflow }}-${{ toJSON(matrix.env) }}-${{ github.ref }}
+      cancel-in-progress: true
+    strategy:
+      fail-fast: false
+      matrix:
+        env:
+          - { CC: "gcc", CFLAGS: "-fno-omit-frame-pointer -fsanitize=undefined -fsanitize=address -ggdb3 -Og -fvar-tracking-assignments" }
+          - { CC: "clang", CFLAGS: "-fno-omit-frame-pointer -fsanitize=undefined -fsanitize=address -ggdb3 -Og" }
```

Maybe

```suggestion
          - { CC: "gcc", CFLAGS: "-fno-omit-frame-pointer -fsanitize=address,undefined -ggdb3 -Og -fvar-tracking-assignments" }
          - { CC: "clang", CFLAGS: "-fno-omit-frame-pointer -fsanitize=address,undefined -ggdb3 -Og" }
```

to make it a little bit shorter.

---

done